vector_indexers/plot.py

```diff
@@ -0,0 +1,68 @@
+import click
+import matplotlib.pyplot as plt
+import json
+import pandas as pd
+
+
+def set_k(recall_at):
+    if recall_at is 1:
+        k = 0
+    elif recall_at is 10:
+        k = 1
+    elif recall_at is 20:
+        k = 2
+    elif recall_at is 50:
+        k = 3
+    elif recall_at is 100:
+        k = 4
+    else:
+        print("K value not valid. It should be either 1, 10, 20, 50 or 100. K = 10 will be used")
+        k = 1
+    return k
+
+
+@click.command()
+@click.option('--recall_at', '-k', default=10)
+def create_plot(recall_at):
+    k = set_k(recall_at)
+
+    input_file = open('results.json')
+    json_array = json.load(input_file)
+
+    result_list = []
+    recall_list = []
+
+    for item in json_array:
+        result_details = {"index_time": None,
+                          "index_docs": None,
+                          "query_time": None,
+                          "build_time": None,
+                          "query_docs": None,
+                          "recall@1,10,20,50,100": None}
+
+        result_details['index_docs'] = item['index_docs']
+        result_details['index_docs'] = item['index_docs']
+        result_details['query_time'] = item['query_time']
+        result_details['build_time'] = item['build_time']
+        result_details['query_docs'] = item['query_docs']
+        result_details['recall@1,10,20,50,100'] = item['recall@1,10,20,50,100']
+        result_list.append(result_details)
+
+    result_list_dataframe = pd.DataFrame(result_list)
+
+    query_time = result_list_dataframe['query_time']
+    recall_at_k = result_list_dataframe['recall@1,10,20,50,100']
+
+    for inner_l in recall_at_k:
```

Comment:
Why are we doing this loop?

I think you should just append the result corresponding to the desired k